components/core/cmake/Toolchains/utils.cmake

```diff
@@ -0,0 +1,40 @@
+function(setup_toolchains)
+    if("Darwin" STREQUAL "${CMAKE_HOST_SYSTEM_NAME}")
+        execute_process(
+            COMMAND
+                sw_vers --productVersion
+            OUTPUT_VARIABLE CMAKE_OSX_VERSION
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        if("${CMAKE_OSX_VERSION}" VERSION_LESS "15")
+            set(CMAKE_TOOLCHAIN_FILE
+                "${CMAKE_TOOLCHAINS_DIR}/llvm-clang-toolchains.cmake"
+                CACHE STRING
+                "Toolchain file"
+            )
+        endif()
+    endif()
+endfunction()
+
+# Require compiler versions that support the C++20 features necessary for compiling CLP
+function(validate_compiler_versions)
+	if("AppleClang" STREQUAL "${CMAKE_CXX_COMPILER_ID}")
+		set(CLP_CMAKE_CXX_COMPILER_MIN_VERSION "16")
```

Comment:
Since functions have their own scope, how about let's simplify the name to `CXX_COMPILER_MIN_VERSION`?